docs/guides/beginner-typescript.md

```diff
@@ -0,0 +1,474 @@
+---
+title: Beginner TypeScript Guide
+nav: 7
+---
+
+Zustand is a lightweight state manager, particularly used with React. Zustand avoids reducers, context, and boilerplate.
+Paired with TypeScript, you get a strongly typed store-state, actions, and selectors-with autocomplete and compile-time safety.
+
+In this basic guide we’ll cover:
+
+- Creating a typed store (state + actions)
+- Using the store in React components with type safety
+- Structuring and coordinating multiple stores
+- Resetting the store safely with types
+- Extracting and reusing Store type (for props, tests, and utilities)
+- Composing multiple selectors and building derived state (with type inference and without extra re-renders)
+- Middlewares with TypeScript support (`persist`, `devtools`, `immer`)
+- Async actions with typed API responses
+- Using `useStore`, `useStoreWithEqualityFn` and `useShallow` hooks with Typescript
+- Additional API (`createStore`, `createWithEqualityFn`)
+
+### Creating a Store with State & Actions
+
+Here we describe state and actions using an Typescript interface. The `<BearState>` generic forces the store to match this shape.
+This means if you forget a field or use the wrong type, TypeScript will complain. Unlike plain JS, this guarantees type-safe state management.
+
+```ts
+import { create } from 'zustand'
+
+// Define types for state & actions
+interface BearState {
+  bears: number
+  food: string
+  increase: (by: number) => void
+  feed: (food: string) => void
+}
+
+// Create store with explicit generic <BearState>
+export const useBearStore = create<BearState>((set) => ({
+  bears: 2,
+  food: 'honey',
+  increase: (by) => set((state) => ({ bears: state.bears + by })),
+  feed: (food) => set(() => ({ food })),
+}))
+```
+
+`create` also supports a curried version. It's useful when you want to fix the state type first and initialize later (for example, inside a factory).
+
+```ts
+import { create } from 'zustand'
+
+// Define types for state & actions
+interface BearState {
+  bears: number
+  food: string
+  increase: (by: number) => void
+  feed: (food: string) => void
+}
+
+// Fix the type with create<BearState>()
+const createBearStore = create<BearState>()
+
+// Initialize the store separately
+export const useBearStore = createBearStore((set) => ({
```

Comment:
We don't recommend separating this. The extra parens `()`  only exists because of TS limitation.

Please stick with `create<T>()(...)` pattern throughout the guide.

```suggestion
export const useBearStore = create<BearState>()((set) => ({
```